File: 명진/programmers/입국 심사.py
def canAfford(n, t, times):
    cnt = 0
    for time in times:
        cnt += t // time
        if cnt >= n:
            return True;
    return False

def solution(n, times):
    l, r = 1, max(times)*n
    while l + 1 < r:
        m = (l + r) // 2
        if (canAfford(n, m, times)):
            r = m
        else:
            l = m
    
    return r

File: 지훈/프로그래머스/3/43238. 입국심사/입국심사.java
class Solution {
    /**
     * Computes the minimum time required to process a given number of tasks using the available processing times.
     * <p>
     * This method employs a binary search algorithm to determine the earliest time at which the cumulative number
     * of tasks processed by all officers (each with their own processing time) meets or exceeds the target count.
     * </p>
     *
     * @param n the total number of tasks to be processed
     * @param times an array where each element represents the processing time for an officer
     * @return the minimum time required to process all tasks
     */
    public long solution(int n, int[] times) {
    
        long left = 1;
        long right = 1_000_000_000L * 100_000L;
        long answer = right;
        
        while (left <= right) {
            long mid = (left + right) / 2;
            
            long total = 0;
            
            for (int time : times) {
                total += mid / time;
            }
            
            if (total >= n) {
                answer = mid;
                right = mid - 1;
            } else {
                left = mid + 1;
            }
            
        }
        
        return answer;
    }
}